program-runtime/src/loaded_programs.rs

```diff
@@ -325,13 +330,25 @@ impl LoadedPrograms {
     /// Evicts programs which were used infrequently
     pub fn sort_and_evict(&mut self, max_cache_entries: Option<usize>) {
         // Find eviction candidates and sort by their usage counters
-        let mut num_cache_entries: usize = 0;
+        let mut num_non_tombstone_entries: usize = 0;
+        let mut num_tombstones = 0;
         let sorted_candidates = self
```

Comment:
The problem with the current eviction strategy is that it only operates on the first level, but it should flatten both levels into one, sort that and then reassemble the two level search indices.